apps/sms/views/conversation/js/conversation.js

```diff
@@ -699,13 +695,24 @@ var ConversationView = {
       Array.from(nodes).forEach((node) => {
         window.URL.revokeObjectURL(node.dataset.thumbnail);
       });
+
+      // If we're leaving conversation view, ensure that the thread object's
+      // unreadCount value is current (set = 0).
+      this.activeThread.unreadCount = 0;
     }
 
     // TODO move most of back() here: Bug 1010223
-    if (this.activeThread &&
+    // We don't want to clean fields, reset active conversation reference or
+    // current draft if user going to report or group view in the scope of the
+    // current conversation.
+    if (!this.activeThread ||
         !this.isConversationPanel(this.activeThread.id, nextPanel)) {
-      // Clean fields when moving out of a conversation.
+      this.updateDraft();
```

Comment:
OK, I think this is the source of the bug I found; I think we should call updateDraft() outside of the if block. What do you think?